pkg/tracer/tracer_linux.go

```diff
@@ -479,6 +567,86 @@ func (t *LinuxTracer) getMap(mapName string) (*bpflib.Map, error) {
 	return mp, nil
 }
 
+func (t *LinuxTracer) dispatchPerfEvent(event *common.PerfEvent) {
+	t.tcpConnInsightsLock.Lock()
+	defer t.tcpConnInsightsLock.Unlock()
+
+	if event.HTTPResponse != nil {
+		logger.Tracef("http response: %v", event.HTTPResponse)
+		httpRes := event.HTTPResponse
+
+		conn, ok := t.tcpConnInsights[httpRes.Connection]
+		httpProtocol := "http"
+		if !ok {
+			conn = ConnInsight{
+				ApplicationProtocol: httpProtocol,
+				HttpMetrics:         make(map[HttpCode]*ddsketch.DDSketch),
+			}
+		}
+		conn.ApplicationProtocol = httpProtocol
+
+		latencyCounter, ok := conn.HttpMetrics[httpRes.StatusCode]
+		if !ok {
+			var err error
+			latencyCounter, err = ddsketch.NewDefaultDDSketch(t.config.HttpMetricPrecision)
+			if err != nil {
+				logger.Errorf("can't create dd sketch")
```

Comment:
This seems like a problem that needs more than logging, because if you just log here, most probably the `latencyCounter` obj would be `nil`, in which case you get a pretty nasty exception on line 579.

Reply:
We've put the code in an else block.